api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminService.java

```diff
@@ -26,4 +27,10 @@ WorkspaceAuditLogQueryResponse getAuditLogEntries(
       Integer limit,
       Long afterMillis,
       @Nullable Long beforeMillisNullable);
+
+  String getReadOnlyNotebook(
+      String workspaceNamespace,
+      String workspaceName,
+      AccessReason accessReason,
```

Comment:
super-nit: I'd kind of expect the notebook name to come before the reason, as it's the third in a line of increasingly specific identifiers.

Reply:
agreed, fixed